subsys/bluetooth/controller/ll_sw/ull_adv_sync.c

```diff
@@ -477,20 +474,61 @@ uint8_t ll_adv_sync_ad_data_set(uint8_t handle, uint8_t op, uint8_t len,
 	}
 #endif /* CONFIG_BT_CTLR_DF_ADV_CTE_TX */
 
-	err = ull_adv_sync_pdu_set_clear(lll_sync, pdu_prev, pdu, ULL_ADV_PDU_HDR_FIELD_AD_DATA, 0,
+	/* Use length = 0 and NULL pointer to retain old data in the PDU.
+	 * Use length = 0 and valid pointer of `data` (auto/local variable) to
+	 * remove old data.
+	 * User length > 0 and valid pointer of `data` (auto/local variable) to
+	 * set new data.
+	 */
+	if (op == BT_HCI_LE_EXT_ADV_OP_UNCHANGED_DATA) {
+		hdr_data[ULL_ADV_HDR_DATA_LEN_OFFSET] = 0U;
+		(void)memset((void *)&hdr_data[ULL_ADV_HDR_DATA_DATA_PTR_OFFSET],
+			     0U, ULL_ADV_HDR_DATA_DATA_PTR_SIZE);
```

Comment:
`memset` actually takes an `int` as input, so `0` may be more "correct" than `0U` in this case, but I can't imagine it mattering 

```suggestion
		(void)memset((void *)&hdr_data[ULL_ADV_HDR_DATA_DATA_PTR_OFFSET],
			     0, ULL_ADV_HDR_DATA_DATA_PTR_SIZE);
```